components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/client/impl/S3ClientIAMOptimizedImpl.java

```diff
@@ -74,12 +74,14 @@ public AmazonS3 getS3Client() {
 
         if (!configuration.isUseEncryption()) {
             clientBuilder = AmazonS3ClientBuilder.standard().withCredentials(new InstanceProfileCredentialsProvider(false));
-        } else if (configuration.isUseEncryption()) {
+
+            if (configuration.hasProxyConfiguration()) {
+                clientBuilder = clientBuilder.withClientConfiguration(clientConfiguration);
+            }
```

Comment:
This seems a slightly different issue.  Should this be in a separate commit ?

Reply:
Absolutely! I can move this change to a separate commit. Let me modify this one and push again.

Reply:
I have just moved this part of the PR to a separate commit.